### ISSUE TITLE
Our password generation functionality suck

### ISSUE THREAD

**Issue by DmitryEfimenko** [openpassphrase/opp-web#151]
I always end up using [this website](https://www.avast.com/en-us/random-password-generator#pc) instead

**Comment by abashmak** [comment]
There are several options that we are not using that would make it better. There's the option to insert randomized special characters or numbers as delimiters between words. And there's an option to randomly capitalize some of the words. We can also very easily add our own custom post-processing to scramble the generated phrase so that it doesn't contain whole words. I think that would make it pretty close to that website. From the UI perspective, perhaps we should only pass in a length argument.

**Comment by abashmak** [comment]
```
from xkcdpass import xkcd_password as xp
from random import shuffle

>>> phrase = xp.generate_xkcdpassword(wordlist,numwords=3,random_delimiters=True,case='random');print(phrase)
3MARBLED@epilogue8THWARTING#
>>> phrase = list(phrase); shuffle(phrase); phrase = ''.join(phrase); print(phrase)
R@NLBg#3EeGAuleoTpAWTDHIRi8M
```
Easy peasy. The length constraint would be a bit trickier to implement, but doable I think, just need to increase numwords and then truncate.

**Comment by DmitryEfimenko** [comment]
any particular reason to do the password generation on the server?

**Comment by abashmak** [comment]
No, we could do it on the client, but no calling out to 3rd party sites, it has to be a client library.

**Comment by DmitryEfimenko** [comment]
do you want to give it a shot?

**Comment by abashmak** [comment]
It would be a much bigger effort for me than changing the backend. There would have to be a compelling reason. I don't see one.

**Comment by DmitryEfimenko** [comment]
Increasing your proficiency in front end?
I'll take this one if you don't want it.

**Comment by abashmak** [comment]
Any particular reason you want to do extra work on the frontend, when backend already has much of the functionality?

**Comment by DmitryEfimenko** [comment]
we'd need to create code to call API to get the password suggestion. An HTTP request for something that does not really need it seems inefficient.

**Comment by abashmak** [comment]
No we don't,  this is already being done by the backend. In fact, we don't need to do anything in the UI and just have the backend choose sensible default parameters for password generation.

**Comment by abashmak** [comment]
Here's the change: https://github.com/openpassphrase/opp/pull/24. It's also deployed live on the demo site. Let's give it a shot and see if it fits our needs.

**Comment by DmitryEfimenko** [comment]
tested on the demo website. Looks good. However, I would like to have an ability to specify the amount of characters.
Recently there was a website where I had to register which required 16 characters for password.
Maybe we could repurpose the current input "Number of words" to "Number of characters".

**Comment by abashmak** [comment]
Yeah, that should be doable. But it seems like an edge use case, do you think you'll hit it that frequently to be worth the effort? The current workaround would be to generate a longer one and truncate it manually.

Let's do it in another PR though. First I'll get the functionality ready but hardcode the length to say 15 chars, then once the frontend is ready to send the length, I'll update to consume it.

**Comment by abashmak** [comment]
TODO: change # of words in the UI to # of characters and implement (currently hard-coded to some fixed length)

**Comment by abashmak** [comment]
Validated, deployed and tested.
Note: `min_length` and `max_length` parameters are actually referring to the min/max lengths of words to pull from the dictionary for password generation, so they remain in available.
